lib/Models/CatalogNew.ts

```diff
@@ -17,13 +17,28 @@ export default class Catalog {
   constructor(terria: Terria) {
     this.terria = terria;
     this.group = new CatalogGroup("/", this.terria);
+    this.terria.addModel(this.group);
+
+    // create user added data group
+    this.userAddedDataGroup;
```

Comment:
We should try to avoid side effects from `get` functions. If we need the group to be created straight away it would be better to create and assign it in the constructor (or a function called by the constructor) and remove `userAddedDataGroupIfItExists`.